tests/phpunit/integration/Core/Util/Feature_FlagsTest.php

```diff
@@ -21,94 +21,47 @@ public function tearDown() {
 		static::reset_feature_flags();
 	}
 
-	public function test_get_mode() {
-		$method = new ReflectionMethod( Feature_Flags::class, 'get_mode' );
-		$method->setAccessible( true );
-		$get_mode = function ( ...$args ) use ( $method ) {
-			return $method->invoke( null, ...$args );
-		};
-
-		// Defaults to 'production'.
-		$this->assertEquals( 'production', $get_mode() );
-
-		// It uses the flag mode it's provided.
-		Feature_Flags::set_mode( 'foo' );
-		$this->assertEquals( 'foo', $get_mode() );
-
-		// Is filterable.
-		$return_custom_mode = function () {
-			return 'custom';
-		};
-		add_filter( 'googlesitekit_flag_mode', $return_custom_mode );
-		$this->assertEquals( 'custom', $get_mode() );
+	public function test_enabled_when_feature_name_is_empty() {
+		remove_all_filters( 'googlesitekit_is_feature_enabled' );
+		Feature_Flags::set_features(
+			array(
+				'test_feature',
+			)
+		);
 
-		// Defaults to production if filter returns falsy.
-		add_filter( 'googlesitekit_flag_mode', '__return_false' );
-		$this->assertEquals( 'production', $get_mode() );
+		$this->assertFalse( Feature_Flags::enabled( '' ) );
 	}
 
-	/**
-	 * @dataProvider data_is_feature_enabled
-	 *
-	 * @param array   $features
-	 * @param string  $mode
-	 * @param string  $feature
-	 * @param boolean $expected
-	 */
-	public function test_enabled( $features, $mode, $feature, $expected ) {
+	public function test_enabled_when_feature_name_is_non_string() {
 		remove_all_filters( 'googlesitekit_is_feature_enabled' );
-		Feature_Flags::set_features( $features );
-		Feature_Flags::set_mode( $mode );
+		Feature_Flags::set_features(
+			array(
+				'test_feature',
+			)
+		);
 
-		if ( $expected ) {
-			$this->assertTrue( Feature_Flags::enabled( $feature ) );
-		} else {
-			$this->assertFalse( Feature_Flags::enabled( $feature ) );
-		}
+		$feature = (object) array( 'foo' => 'bar' );
+		$this->assertFalse( Feature_Flags::enabled( $feature ) );
 	}
 
-	public function data_is_feature_enabled() {
-		return array(
-			'no feature given'                             => array(
-				array(
-					'test_feature' => 'production',
-				),
-				'production',
-				'',
-				false,
-			),
-			'non-string truthy feature given'              => array(
-				array(
-					'test_feature' => 'production',
-				),
-				'production',
-				(object) array( 'foo' => 'bar' ),
-				false,
-			),
-			'feature enabled for production in production' => array(
-				array(
-					'test_feature' => 'production',
-				),
-				'production',
-				'test_feature',
-				true,
-			),
-			'feature enabled for development in production' => array(
-				array(
-					'test_feature' => 'development',
-				),
-				'production',
-				'test_feature',
-				false,
-			),
-			'feature enabled for development and production in development' => array(
-				array(
-					'test_feature' => array( 'development', 'production' ),
-				),
-				'development',
+	public function test_enabled_with_filter() {
```

Comment:
It seems the current tests here would pass or fail the same way if the method only checked that the feature is in the list of available features.

Let's add another test to show that features are not enabled by default. i.e. same as this one where the feature is known and the key is valid but no added filter to enable.

Reply:
Updated.